aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -1776,7 +1789,55 @@ std::optional<c10::ScalarType> out_dtype) {
 
   Tensor out = create_grouped_gemm_output_tensor(mat_a, mat_b, offs, out_dtype);
 
-  at::cuda::detail::bf16bf16_grouped_mm(mat_a, mat_b, offs, bias, out);
+  // TODO(before land): clean up the if statement, naming, etc
+  if (use_fast_path) {
+    // fast path, no d2h sync needed
+    at::cuda::detail::bf16bf16_grouped_mm(mat_a, mat_b, offs, bias, out);
+  } else {
+    // fallback path, using for loops or bmm
+    TORCH_WARN("fallback path for `torch._grouped_mm`, performance may not be optimal");
```

Comment:
make this LOG(INFO)

Reply:
sure, did this in next PR in the stack (https://github.com/pytorch/pytorch/pull/161717)